static/js/recent_senders.js

```diff
@@ -238,6 +238,11 @@ export function get_pm_recent_senders(user_ids_string) {
         return max_id2 - max_id1;
     }
 
+    // PM to self.
+    if (user_ids.length === 1 && user_ids[0] === people.my_current_user_id()) {
+        pm_senders_info.participants.push(user_ids[0]);
+        return pm_senders_info;
+    }
     // Add current user to user_ids.
     user_ids.push(people.my_current_user_id());
```

Comment:
The early return makes this feel a bit hard to read.

Should we instead just make this `push` conditional on that user ID not already being present... or maybe also do this earlier when constructing `user_ids` at the top. Or wait, how does this early return even affect participants? I'm a bit confused.

Reply:
At this point, `pm_senders_info` is empty and we know for pm to self, the only participant is the user. We don't need to do any of the sorting done below too. This is a special case, so I am not sure if we should make processing for other cases more expensive just to handle this one.